docs/website/docs/general-usage/rest-client.md

```diff
@@ -0,0 +1,259 @@
+# Reading data from RESTful APIs
+
+RESTful APIs are a common way to interact with web services. They are based on the HTTP protocol and are used to read and write data from and to a web server. dlt provides a simple way to read data from RESTful APIs using two helper methods: [a wrapper around the `requests`](../reference/performance#using-the-built-in-requests-client) library and a `RESTClient` class.
+
+:::tip
+There's also shorthand function to read from paginated APIs. Check out the [paginate()](#shortcut-for-paginating-api-responses) function.
+:::
+
+
+The `RESTClient` class offers a powerful interface for interacting with RESTful APIs, supporting features like
+- automatic pagination,
+- various authentication mechanisms,
+- customizable request/response handling.
+
+This guide demonstrates how to use the `RESTClient` class to read data APIs focusing on its `paginate()` method to fetch data from paginated API responses.
+
+## Quick example
+
+Here's a simple pipeline that reads issues from the [dlt GitHub repository](https://github.com/dlt-hub/dlt/issues). The API endpoint is `https://api.github.com/repos/dlt-hub/dlt/issues`. The result is "paginated", meaning that the API returns a limited number of issues per page. The `paginate()` iterates over all pages and yields the results which are then processed by the pipeline.
```

Comment:
I would add that in this example authentification and pagination are guessed automatically. And after this example put another one that shows how you can specify the paginator and auth if they are special/not guessed correctly.

Honestly, these two examples would be enough to give users an understanding of how to use RESTClient.

So overall I suggest:
* Quick example simple
* Quick example with specific paginator/auth
* Available Paginators
* Available Auths
* How to write my own if nothing fits

In this case user WOW with this doc would be: 
1) Copy-past code which guesses everything automatically 
2) If it doesn't work, copy-past code with a specification of pagination and auth.
3) Search in docs which auth and paginator is right for you
4) If nothing fits -> How-to write custom auth/paginator.